alignak/acknowledge.py

```diff
@@ -101,34 +99,16 @@ def __init__(self, ref, sticky, notify, persistent,
         self.comment = comment
         self.persistent = persistent
 
-    def __getstate__(self):
-        """Call by pickle for dataify the acknowledge
-        because we DO NOT WANT REF in this pickleisation!
+    def serialize(self):
+        """This function serialize into a simple dict object.
+        It is used when transferring data to other daemons over the network (http)
 
-        :return: dictionary of properties
-        :rtype: dict
-        """
-        cls = self.__class__
-        # id is not in *_properties
-        res = {'_id': self._id}
-        for prop in cls.properties:
-            if hasattr(self, prop):
-                res[prop] = getattr(self, prop)
-        return res
+        Here we directly return all attributes
 
-    def __setstate__(self, state):
-        """
-        Inverse function of getstate
-
-        :param state: it's the state
-        :type state: dict
-        :return: None
+        :return: json representation of a Acknowledge
+        :rtype: dict
         """
-        cls = self.__class__
-        self._id = state['_id']
-        for prop in cls.properties:
-            if prop in state:
-                setattr(self, prop, state[prop])
-        # If load a old ack, set the end_time to 0 which refers to infinite
-        if not hasattr(self, 'end_time'):
-            self.end_time = 0
+        return {'uuid': self.uuid, 'ref': self.ref, 'sticky': self.sticky, 'notify': self.notify,
```

Comment:
The self.ref is the Schedulingitem (host for example), so you put in the serialization a complete object. Not sure we can ujson this dict. Perhaps need to add in this the object name + the UUID